Lib/test/_test_multiprocessing.py

```diff
@@ -5760,6 +5760,33 @@ def test_namespace(self):
         self.run_worker(self._test_namespace, o)
 
 
+class TestNamedResource(unittest.TestCase):
+    def test_global_named_resource_spawn(self):
+        #
+        # Check that global named resources in main module
+        # will not leak by a subprocess, in spawn context.
+        #
+        with os_helper.temp_dir() as tmp_dir:
+            source = os.path.join(tmp_dir, 'source.py')
+            with open(source, 'w') as f:
+                f.write('''if 1:
+                    import multiprocessing as mp
+
+                    ctx = mp.get_context('spawn')
+
+                    global_resource = ctx.Semaphore()
+
+                    def submain(): pass
+
+                    if __name__ == '__main__':
+                        p = ctx.Process(target=submain)
+                        p.start()
+                        p.join()
+                ''')
+            rc, out, err = test.support.script_helper.assert_python_ok(source)
+            self.assertNotRegex(err, b'resource_tracker: There appear to be .* leaked')
```

Comment:
Should we just check that `err` is empty here?

Reply:
I agree with @pitrou.

Reply:
Yep.